File: StudentOglasi/StudentOglasi.Services/Database/Rezervacije.cs
using System;
using System.Collections.Generic;

namespace StudentOglasi.Services.Database;

public partial class Rezervacije
{
    public int Id { get; set; }

    public int SmjestajnaJedinicaId { get; set; }

    public int StudentId { get; set; }

    public DateTime DatumPrijave { get; set; }

    public DateTime DatumOdjave { get; set; }

    public int? BrojOsoba { get; set; }

    public decimal Cijena { get; set; }

    public string? Napomena { get; set; }

    public int? StatusId { get; set; }

    public virtual SmjestajnaJedinica SmjestajnaJedinica { get; set; } = null!;

    public virtual StatusPrijave? Status { get; set; }

    public virtual Studenti Student { get; set; } = null!;
}

File: StudentOglasi/UI/studentoglasi_mobile/lib/models/Rezervacije/rezervacije.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:studentoglasi_mobile/models/Smjestaj/smjestaj.dart';
import 'package:studentoglasi_mobile/models/SmjestajnaJedinica/smjestajna_jedinica.dart';
import 'package:studentoglasi_mobile/models/StatusPrijave/statusprijave.dart';
import 'package:studentoglasi_mobile/models/Student/student.dart';

part 'rezervacije.g.dart';

@JsonSerializable()
class Rezervacije {
  int? studentId;
  int? smjestajnaJedinicaId;
  DateTime? datumPrijave;
  DateTime? datumOdjave;
  int? brojOsoba;
  String? napomena;
  double? cijena;
  int? statusId;
  SmjestajnaJedinica? smjestajnaJedinica;
  Smjestaj? smjestaj;
  StatusPrijave? status;
  Student? student;

  Rezervacije(
      this.studentId,
      this.smjestajnaJedinicaId,
      this.datumPrijave,
      this.datumOdjave,
      this.brojOsoba,
      this.napomena,
      this.cijena,
      this.statusId,
      this.smjestajnaJedinica,
      this.status,
      this.student,
      this.smjestaj);
  factory Rezervacije.fromJson(Map<String, dynamic> json) =>
      _$RezervacijeFromJson(json);

  Map<String, dynamic> toJson() => _$RezervacijeToJson(this);
}
